profiles: # definitions for constraint profiles
  default: # default profile must exist
    allow_reverse: True
    L: 0.29
    speed_limit: 0.4
    steer_limit: 0.35 # 0.1 * pi
    # To calculate turning radius, use r = L / tan(abs(steer_limit)))
    penaltyTurning: 1.0 #100
    penaltyHardTurning: 1.0 #100
    penaltyReversing: 200.0
    penaltyCOD: 2.0
    heuristicWeight: 2.0
    mapResolution: 0.1
    carWidth: 0.35 # 0.27
    LF: 0.40 # 0.38
    LB: 0.1 # 0.08
    obsRadius: 0.1
    constraintWaitTime: 2
    space_buffer: 0.0 # buffer is relative to the car's dimensions
  pushing: # non-default profiles only need to specify non-default values
    allow_reverse: False
    steer_limit: 0.18
    LF: 0.50 # 0.58 # includes length of block being pushed

# which profile to use when navigating to each waypoint (0 indexed)
# if a profile is not specified for a waypoint, default profile is used
profile0: "default"
profile1: "pushing"
profile2: "default"
profile3: "pushing"
